### PR TITLE
A simpler version of the Semigroup instance for MaybeDefault

### PR DESCRIPTION
**Pull request by KommuSoft**
The implementation of the `Semigroup` has less clauses, but I think it is also more descriptive since the code "explains" how the semigroup works whereas in the original version it exhaustively enumerated over the items, but without explaining the rules of the semigroup.

Likely the implementation is also slightly more memory efficient, since we return the `SetTo _` parameter, not constructing a new equivalent one. The implementation is also more lazy, since we never evaluate the first parameter here. If the second item is thus `Default` or `SetTo _` we will never evaluate the first parameter.

A test is created to check if the implementation is equivalent to the old `Semigroup` instance.

### CODE REVIEW

File: src/Graphics/Vty/Attributes.hs [hunk 1]
```diff
@@ -154,7 +154,7 @@ instance (NFData v) => NFData (MaybeDefault v) where
 instance Eq v => Semigroup (MaybeDefault v) where
     _        <> v@(SetTo _) = v
     x        <> KeepCurrent = x
-    _        <> Default     = Default
+    _        <> ~Default    = Default
```
Comment (jtdaugherty) [inline src/Graphics/Vty/Attributes.hs]: Please revert this change.

Comment (KommuSoft) [reply src/Graphics/Vty/Attributes.hs]: Removed the irrefutable pattern.

**Comment by jtdaugherty** [comment]
Thank you!
